lib/state_sampler.h

```diff
@@ -0,0 +1,69 @@
+#ifndef STATE_SAMPLER_H_
+#define STATE_SAMPLER_H_
+
+#include <complex>
+#include <memory>
+#include <random>
+
+namespace qsim {
+
+template <typename StateSpace>
+class StateSampler {
+ public:
+  using State = typename StateSpace::State;
+
+  // Sample using a random seed.
+  void SampleState(const StateSpace& statespace, const State& state,
```

Comment:
I think such a function shouldn't be provided as it leads to unreproducible results.